inc/class-s3-uploads.php

```diff
@@ -277,4 +352,163 @@ public function copy_image_from_s3( $file ) {
 		copy( $file, $temp_filename );
 		return $temp_filename;
 	}
+
+	/**
+	 * Check if the attachment is private.
+	 *
+	 * @param integer $attachment_id
+	 * @return boolean
+	 */
+	public function is_private_attachment( int $attachment_id ) : bool {
+		/**
+		 * Filters whether an attachment should be private.
+		 *
+		 * @param bool Whether the attachment is private.
+		 * @param int  The attachment ID.
+		 */
+		return apply_filters( 's3_uploads_is_attachment_private', false, $attachment_id );
+	}
+
+	/**
+	 * Update the ACL (Access Control List) for an attachments files.
+	 *
+	 * @param integer $attachment_id
+	 * @param string $acl public-read|private
+	 * @return void
+	 */
+	public function set_attachment_files_acl( int $attachment_id, string $acl ) : ?WP_Error {
+		$files = static::get_attachment_files( $attachment_id );
+		$locations = array_map( [ $this, 'get_s3_location_for_path' ], $files );
+		$s3 = $this->s3();
+		$commands = [];
+		foreach ( $locations as $location ) {
+			$commands[] = $s3->getCommand( 'putObjectAcl', [
+				'Bucket' => $location['bucket'],
+				'Key' => $location['key'],
+				'ACL' => $acl,
+			] );
+			try {
```

Comment:
Should the try/catch block be outside of foreach? Right now it seems to be executing the CommandPool on each command individually as soon as the command is added to the commands array.

Reply:
Whoops! nice catch :D